text/0000-feature-metadata-doc-cfg.md

```diff
@@ -0,0 +1,334 @@
+- Feature Name: feature-metadata-doc-cfg
+- Start Date: 2023-04-14
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/3416)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC has three simple goals:
+
+1. Add a way to write feature descriptions in `Cargo.toml`, as well some other
+   simple feature attributes.
+2. Establish a way for Cargo or other build systems to easily pass advanced
+   configuration information to `rustdoc` (e.g., favicon or playground URLs from
+   `Cargo.toml`)
+3. Allow `rustdoc` to accept this information either from the CLI or from a new
+   `rustdoc.toml` file.
```

Comment:
I was unsure whether it would be good to split RFCs or keep them together, since cargo and rustdoc definitely do need to be in sync here. I think I'll leave it as-is for a few days, just to promote some joint discussion, then split into two separate RFCs to expand on the details.

Reply:
When an RFC covers too much, it is susceptible to
- The discussion being harder to close out because every round of updates someone goes back and finds something they didn't notice on a previous pass
- The discussion being larger, generally, and is harder to follow, making it more likely for parts to fall through the cracks
- Details being missed because people's focus is on the more contentious items

Reply:
@epage what sort of use case were you thinking for `cargo add`? I'm in the process of splitting RFCs, and will add it as an example

Reply:
`cargo add` lists the features for the dependency being added.  We could list out the descriptions too.

Which brings us to semantics.  Is `doc` like `package.description` where its content is more free form (I think?).  Or is it more like doc-comments where its markdown and we can exctract a one-line summary for use in places like `cargo add`?

Reply:
In the rewrite I have stated that it’s like the equivalent of `#[doc(…)]` with the first line being a summary, but I this can of course be change (I should have the updated RFCs here by tomorrow)

Reply:
I have updated the use cases in this RFC, which is now only for `Cargo.toml`